examples/mig_stateful/main.tf

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+provider "google" {
+  project = var.project_id
+  region  = var.region
+}
+
+provider "google-beta" {
+  project = var.project_id
+  region  = var.region
+}
+
+resource "random_string" "suffix" {
+  length  = 4
+  special = "false"
+  upper   = "false"
+}
+
+/** Network **/
+
+resource "google_compute_network" "main" {
+  project                 = var.project_id
+  name                    = "cft-vm-test-${random_string.suffix.result}"
+  auto_create_subnetworks = "false"
+}
+
+resource "google_compute_subnetwork" "main" {
+  project       = var.project_id
+  region        = var.region
+  name          = "cft-vm-test-${random_string.suffix.result}"
+  ip_cidr_range = "10.128.0.0/20"
+  network       = google_compute_network.main.self_link
+}
+
+module "instance_template" {
+  source             = "../../modules/instance_template"
+  project_id         = var.project_id
+  subnetwork         = var.subnetwork
+  service_account    = var.service_account
+  subnetwork_project = var.project_id
+}
+
+module "mig" {
+  source            = "../../modules/mig"
+  project_id        = var.project_id
+  region            = var.region
+  target_size       = var.target_size
+  hostname          = "mig-stateful"
+  instance_template = module.instance_template.self_link
+  stateful_ips = [{
+    interface_name = "nic0"
+    delete_rule    = "ON_PERMANENT_INSTANCE_DELETION"
+    is_external    = true
+  }]
+
+  update_policy = [{
+    max_surge_fixed              = 0
+    instance_redistribution_type = "NONE"
+    max_surge_percent            = null
+    max_unavailable_fixed        = 4
+    max_unavailable_percent      = null
+    min_ready_sec                = 180
+    replacement_method           = "RECREATE"
+    minimal_action               = "REFRESH"
+    type                         = "OPPORTUNISTIC"
+  }]
```

Comment:
I don't see any assertions for this in your test. I would suggest either take these (and other) superfluous configs out of the example, if they are not critical for the example OR add assertions in your test if they are necessary.